### PR TITLE
Added copying of headers to the fronted request closes #4246

### LINKED ISSUE #4246: DDF requests not including caching headers

### PR DESCRIPTION
**Pull request by myleshorton**

### CODE REVIEW

**Comment by myleshorton** [comment]
OK @oxtoacart I think this should be all set. I went ahead and refactored things a little bit to make testing easier.

**Comment by myleshorton** [comment]
Hey one sec @oxtoacart just realized the test needs a little work..

**Comment by oxtoacart** [comment]
@myleshorton Ok, just let me know when you want me to take a look.

File: src/github.com/getlantern/flashlight/util/http.go [hunk 1]
```diff
@@ -94,19 +95,48 @@ type dualFetcher struct {
...
 	log.Debugf("Using dual fronter")
-	frontedUrl := req.Header.Get("Lantern-Fronted-URL")
+	frontedURL := req.Header.Get("Lantern-Fronted-URL")
 	req.Header.Del("Lantern-Fronted-URL")
 
-	if frontedUrl == "" {
+	// Make a copy of the original requeest headers to include in the fronted
+	// request. This will ensure that things like the caching headers are
+	// included in both requests.
+	headersCopy := make(http.Header, len(req.Header))
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/util/http.go]: Small nit, but there's no point in copying if the frontedURL check fails.

**Pull request merged**